dig_test.go

```diff
@@ -883,17 +883,14 @@ func TestCantProvideParameterObjects(t *testing.T) {
 	})
 
 	t.Run("pointer from constructor", func(t *testing.T) {
+		c := New()
 		type Args struct{ In }
 
 		args := &Args{}
 
-		c := New()
-		require.NoError(t, c.Provide(func() (*Args, error) {
-			return args, nil
-		}), "provide failed")
-		require.NoError(t, c.Invoke(func(a *Args) {
-			require.True(t, args == a, "args must match")
-		}), "invoke failed")
+		err := c.Provide(func() (*Args, error) { return args, nil })
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "can't provide pointers to parameter objects")
```

Comment:
This test has somehow gotten weird - why are we writing test cases that *return* `fx.In`? I think that the intention was to test that we can write constructors that return pointers. If that's not covered elsewhere, let's add a test with a constructor that returns `*byte.Buffer`.

Separately, we should add whatever test we need for pointers to In/Out structs.